### PR TITLE
Bump `github.com/gardener/gardener` to `v1.89.0`

### PR DESCRIPTION
**Pull request by plkokanov**
**How to categorize this PR?**

/kind enhancement

**What this PR does / why we need it**:
This PR updates the github.com/gardener/gardener dependency to v1.86.0. Additionally, it removes the `-mod=vendor` option from the testmachinery [`test-definition`](https://github.com/plkokanov/gardener-extension-shoot-rsyslog-relp/blob/23e5442cbd6e3ab567c2ecf25900fbcd78c23130/.test-defs/TestSuiteShootRsyslogRelpBetaSerial.yaml#L15) as the vendor folder was already removed.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Bumped `github.com/gardener/gardener` to `v1.89.0.`
```
```other developer
Bumped golang to `v1.22.0`
```

### CODE REVIEW

File: .ci/hack/component_descriptor [hunk 1]
```diff
@@ -1,10 +1,14 @@
 #!/usr/bin/env bash
 
+# SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
```
Comment (ialidzhikov) [inline .ci/hack/component_descriptor:3]: https://github.com/gardener/gardener-extension-shoot-rsyslog-relp/blob/8490de8f362951b0b34730a6a32c386852b23dee/.ci/component_descriptor#L7-L9: there is one TODO can be addressed with this vendoring.

**Comment by ialidzhikov** [comment]
Looks good apart from the failing `make generate`:
```
>> make generate
make generate needs to be run:
 M imagevector/images.go
```

**Comment by plkokanov** [comment]
The `make generate` was failing because of the `generate-imagename-constants.sh` script, which is used to generate constants for the images used by the extension, now adds `reuse` license headers to the generated files. 
Additionally, I noticed a misplaced import and addressed that aswell.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**